### PR TITLE
chore: apps start using protobuf payloads

### LINKED ISSUE #12850: Apps start using protobuf payloads

### PR DESCRIPTION
**Pull request by lpetrovic05**
closes #12850

### CODE REVIEW

**Review by cody-littley — APPROVED** [review]

File: platform-sdk/platform-apps/demos/CryptocurrencyDemo/src/main/java/com/swirlds/demo/crypto/CryptocurrencyDemoState.java [hunk 1]
```diff
@@ -227,21 +226,22 @@ public void handleConsensusRound(final Round round, final PlatformState swirldDu
      * {ASK,s,p} = ask to sell 1 share of stock s at p cents (where 1 &lt;= p &lt;= 127)
      * </pre>
      */
-    private void handleTransaction(
-            @NonNull final NodeId id, final boolean isConsensus, @Nullable final Transaction transaction) {
+    private void handleTransaction(@NonNull final NodeId id, @NonNull final Transaction transaction) {
         Objects.requireNonNull(id, "id must not be null");
-        if (transaction == null || transaction.getContents().length == 0) {
+        Objects.requireNonNull(id, "transaction must not be null");
```
Comment (anthony-swirldslabs) [inline platform-sdk/platform-apps/demos/CryptocurrencyDemo/src/main/java/com/swirlds/demo/crypto/CryptocurrencyDemoState.java:231]: `s/id/transaction/` in the first argument.

File: platform-sdk/platform-apps/tests/StressTestingTool/src/main/java/com/swirlds/demo/stress/StressTestingToolState.java [hunk 2]
```diff
@@ -102,7 +102,10 @@ public void handleConsensusRound(final Round round, final PlatformState platform
     }
 
     private void handleTransaction(@NonNull final ConsensusTransaction trans) {
-        runningSum += ByteUtils.byteArrayToLong(trans.getContents(), 0);
+        if (trans.isSystem()) {
+            return;
+        }
+        runningSum += ByteUtils.byteArrayToLong(trans.getApplicationPayload().toByteArray(), 0);
```
Comment (anthony-swirldslabs) [inline platform-sdk/platform-apps/tests/StressTestingTool/src/main/java/com/swirlds/demo/stress/StressTestingToolState.java:108]: Minor: note that `toByteArray()` creates a new primitive bytes array. While it's totally fine in demo code, if you find a need to use a similar call in production code, then it might be worth implementing `Bytes.toLong()` to reduce the load on the GC.

File: platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolState.java [hunk 3]
```diff
@@ -223,7 +223,11 @@ public void handleConsensusRound(@NonNull final Round round, @NonNull final Plat
      * @param transaction the transaction to apply
      */
     private void handleTransaction(@NonNull final ConsensusTransaction transaction) {
-        final int delta = ByteUtils.byteArrayToInt(transaction.getContents(), 0);
+        if (transaction.isSystem()) {
```
Comment (alittley) [inline platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolState.java:226]: You're ignoring system transactions in several places, but it's not immediately clear from the code why that should be the case

File: platform-sdk/platform-apps/tests/ConsistencyTestingTool/src/test/java/com/swirlds/demo/consistency/ConsistencyTestingToolRoundTests.java [hunk 4]
```diff
@@ -32,6 +34,7 @@
 import com.swirlds.platform.system.address.AddressBook;
 import com.swirlds.platform.system.transaction.ConsensusTransaction;
 import com.swirlds.platform.system.transaction.ConsensusTransactionImpl;
+import com.swirlds.proto.event.EventPayload.PayloadOneOfType;
```
Comment (alittley) [inline platform-sdk/platform-apps/tests/ConsistencyTestingTool/src/test/java/com/swirlds/demo/consistency/ConsistencyTestingToolRoundTests.java:37]: Where does this package name come from? I think the correct way to import this might be like this:

```
import com.hedera.hapi.platform.event.EventPayload.PayloadOneOfType;
```

File: platform-sdk/platform-apps/tests/MigrationTestingTool/src/main/java/com/swirlds/demo/migration/TransactionUtils.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+package com.swirlds.demo.migration;
+
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.swirlds.common.io.streams.SerializableDataInputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+
+public class TransactionUtils {
```
Comment (alittley) [inline platform-sdk/platform-apps/tests/MigrationTestingTool/src/main/java/com/swirlds/demo/migration/TransactionUtils.java:24]: Missing class doc

File: platform-sdk/platform-apps/tests/MigrationTestingTool/src/main/java/com/swirlds/demo/migration/TransactionUtils.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    /**
+     * Parse a {@link MigrationTestingToolTransaction} from a {@link Bytes}.
+     */
+    public static MigrationTestingToolTransaction parseTransaction(final Bytes bytes) {
```
Comment (alittley) [inline platform-sdk/platform-apps/tests/MigrationTestingTool/src/main/java/com/swirlds/demo/migration/TransactionUtils.java:28]: missing nullity annotations for return value, and `bytes` parameter

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/transaction/Transaction.java [hunk 7]
```diff
@@ -53,6 +55,16 @@ public sealed interface Transaction extends SerializableWithKnownLength permits
     @NonNull
     OneOf<EventPayload.PayloadOneOfType> getPayload();
 
+    /**
+     * A convenience method for retrieving the application payload {@link Bytes} object. Before calling this method,
+     * ensure that the transaction is not a system transaction by calling {@link #isSystem()}.
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/transaction/Transaction.java:60]: > Before calling this method, ensure that the transaction is not a system transaction by calling {@link #isSystem()}.

Why is this necessary, since the method does this check internally?

Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/transaction/Transaction.java:60]: Because you might expect to get a payload but will get empty Bytes

File: platform-sdk/platform-apps/demos/CryptocurrencyDemo/src/main/java/com/swirlds/demo/crypto/CryptocurrencyDemoState.java (hunk 1 above)
Comment (lpetrovic05) [reply platform-sdk/platform-apps/demos/CryptocurrencyDemo/src/main/java/com/swirlds/demo/crypto/CryptocurrencyDemoState.java:231]: fixed

File: platform-sdk/platform-apps/tests/StressTestingTool/src/main/java/com/swirlds/demo/stress/StressTestingToolState.java (hunk 2 above)
Comment (lpetrovic05) [reply platform-sdk/platform-apps/tests/StressTestingTool/src/main/java/com/swirlds/demo/stress/StressTestingToolState.java:108]: yeah, I know. this PR is just for demo apps, didn't want to spend too much time on this since these apps don't even have unit tests. for production, we will not copy

File: platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolState.java (hunk 3 above)
Comment (lpetrovic05) [reply platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolState.java:226]: when the whole project is done, we are supposed to expose system transactions to apps. Hedere will write them to the block stream. In preparation for that, the demo apps will ignore these

File: platform-sdk/platform-apps/tests/ConsistencyTestingTool/src/test/java/com/swirlds/demo/consistency/ConsistencyTestingToolRoundTests.java (hunk 4 above)
Comment (lpetrovic05) [reply platform-sdk/platform-apps/tests/ConsistencyTestingTool/src/test/java/com/swirlds/demo/consistency/ConsistencyTestingToolRoundTests.java:37]: it is generated based on the `.proto` file:
```
option java_package = "com.swirlds.proto.legacy.event";
// <<<pbj.java_package = "com.swirlds.proto.event">>> This comment is special code for setting PBJ Compiler java package
```

File: platform-sdk/platform-apps/tests/MigrationTestingTool/src/main/java/com/swirlds/demo/migration/TransactionUtils.java (hunk 5 above)
Comment (lpetrovic05) [reply platform-sdk/platform-apps/tests/MigrationTestingTool/src/main/java/com/swirlds/demo/migration/TransactionUtils.java:24]: added

File: platform-sdk/platform-apps/tests/MigrationTestingTool/src/main/java/com/swirlds/demo/migration/TransactionUtils.java (hunk 6 above)
Comment (lpetrovic05) [reply platform-sdk/platform-apps/tests/MigrationTestingTool/src/main/java/com/swirlds/demo/migration/TransactionUtils.java:28]: added

**Review by alittley — APPROVED** [review]

**Review by anthony-swirldslabs — DISMISSED** [review]

**Review by mxtartaglia-sl — DISMISSED** [review]

**Review by jeromy-cannon — DISMISSED** [review]

**Review by cody-littley — APPROVED** [review]

**Review by alittley — APPROVED** [review]
